### PR TITLE
Adds a script for closing old issues on the repo

### PR DESCRIPTION
**Pull request by orta**
Issues which are closed:

- Anything which hasn't had a comment in over 6 months 
- If the issue has _any_ labels then it is skipped

I validated in https://github.com/orta/FakeDTRepo/issues/2 that commenting updates the `updated_at` which is used in the search query. This doesn't add any new deps, it's already in the tree. I just make it explicit.

This is the query results: https://github.com/DefinitelyTyped/DefinitelyTyped/search?q=type%3Aissue+updated%3A%3C%3D2020-12-04+repo%3ADefinitelyTyped%2FDefinitelyTyped+state%3Aopen&type=issues

### CODE REVIEW

**Review by RyanCavanaugh — APPROVED** [review]

File: scripts/close-old-issues.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    // For example:
+    // https://github.com/DefinitelyTyped/DefinitelyTyped/search?q=updated%3A%3C%3D2021-02-01&state=open&type=issues
+
+    const parameters = {
+        per_page: 100,
+        q: `type:issue updated:<=${dateQuery} repo:DefinitelyTyped/DefinitelyTyped state:open`
+    };
+
+    for await (const response of octokit.paginate.iterator('GET /search/issues', parameters)) {
+        debugger
```
Comment (SimonSchick) [inline scripts/close-old-issues.js:26]: You probably want to remove that

File: scripts/close-old-issues.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            await octokit.issues.createComment({ repo: "DefinitelyTyped", owner: "DefinitelyTyped", number: issue.number, body: message })
+        }
+    }
+};
+
+go();
+
+// https://stackoverflow.com/questions/1648392/get-a-date-object-six-months-prior-from-another-date-object
+function addMonths(date, months) {
+    var month = (date.getMonth() + months) % 12;
```
Comment (SimonSchick) [inline scripts/close-old-issues.js:43]: Can't you just use `date.setMonth(date.getMonth() + 1)` since JS dates automatically roll over?

Comment (orta) [reply scripts/close-old-issues.js:43]: That was my original code, but the reasoning for this is explained in the Stack Overflow link

File: scripts/close-old-issues.js (hunk 1 above)
Comment (orta) [reply scripts/close-old-issues.js]: Thanks!

**Pull request merged**